staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -230,45 +151,51 @@ func NewRequestManagementSystem(
 	return reqMgmt
 }
 
-// Run bootstraps the request-management system
-func (r *requestManagementSystem) Run(stopCh <-chan struct{}) error {
+// Run runs the controller that deals with config API objects
+func (reqMgmt *requestManagementSystem) Run(stopCh <-chan struct{}) error {
 	// TODO: implement
 	return nil
 }
 
-// GetCurrentState returns the current state of the request-management system
-func (r *requestManagementSystem) GetCurrentState() *RequestManagementState {
-	return r.curState.Load().(*RequestManagementState)
-}
-
-// GetFairQueuingSystem returns the fair-queuing system of the priority-level
-func (s *PriorityLevelState) GetFairQueuingSystem() FairQueuingSystem {
-	return s.fairQueuingSystem
-}
-
 // RequestDigest holds necessary info from request for flow-control
 type RequestDigest struct {
 	RequestInfo *request.RequestInfo
 	User        user.Info
 }
 
-// PickFlowSchema returns a best-matching flow-schema according to the request
-func PickFlowSchema(digest RequestDigest, flowSchemas FlowSchemaSequence, priorityLevelStates map[string]*PriorityLevelState) *rmtypesv1alpha1.FlowSchema {
-	// TODO: implement
-	return nil
+func (reqMgmt *requestManagementSystem) Wait(requestDigest RequestDigest) (execute bool, afterExecute func()) {
+	for {
+		rmState := reqMgmt.curState.Load().(*RequestManagementState)
+		fs := rmState.pickFlowSchema(requestDigest)
+		// RequestManagementState is constructed to guarantee that the following succeeds
+		ps := rmState.priorityLevelStates[fs.Spec.PriorityLevelConfiguration.Name]
+		if ps.config.Exempt {
+			klog.V(5).Infof("Serving %v without delay", requestDigest)
```

Comment:
nit: in favor of V7? above V8 we start dumping request into logs, at V7 we can debug the system w/o filtering extra request payloads

Reply:
Is there any general guidance on how 5 differs from 7?